dagster_polars/io_managers/parquet.py

```diff
@@ -74,30 +51,26 @@ def scan_parquet(path: "UPath", context: InputContext) -> pl.LazyFrame:
     :param context:
     :return:
     """
-    assert context.metadata is not None
-
-    storage_options: Optional[dict[str, Any]] = None
+    context_metadata = context.metadata or {}
 
-    try:
-        storage_options = path.storage_options
-    except AttributeError:
-        # TODO: explore removing this as universal-pathlib should always provide storage_options in newer versions
-        pass
+    storage_options: Optional[dict[str, Any]] = path.storage_options if hasattr(path, "storage_options") else None
 
     kwargs = dict(
-        n_rows=context.metadata.get("n_rows", None),
-        cache=context.metadata.get("cache", True),
-        parallel=context.metadata.get("parallel", "auto"),
-        rechunk=context.metadata.get("rechunk", True),
-        row_count_name=context.metadata.get("row_count_name", None),
-        row_count_offset=context.metadata.get("row_count_offset", 0),
-        low_memory=context.metadata.get("low_memory", False),
-        use_statistics=context.metadata.get("use_statistics", True),
+        n_rows=context_metadata.get("n_rows", None),
+        cache=context_metadata.get("cache", True),
+        parallel=context_metadata.get("parallel", "auto"),
+        rechunk=context_metadata.get("rechunk", True),
+        low_memory=context_metadata.get("low_memory", False),
```

Comment:
I think we want to be able to set options like this from the IOManager config too (not just from specific asset metadata). 

By the way, perhaps we can have a helper function like: 

```python
def get_from_metadata_or_config(self, context: Union[InputContext, OutputContext], param: str, default):
    return context.metadata.get(param) or getattr(self, param, default)
```

this would remove a lot of repetitive code

Reply:
@danielgafni let's do that in another refactor, otherwise we deviate to much from the scope of this PR.

Reply:
ok